lib/tapi_rpc/misc.c

```diff
@@ -3043,14 +3043,45 @@ te_bool
 tapi_interface_is_vlan(rcf_rpc_server *rpcs,
                        const struct if_nameindex *interface)
 {
-    char if_par[IFNAMSIZ] = {0};
+    char     if_name[IFNAMSIZ];
+    char    *val = NULL;
+    te_errno rc;
 
-    rpc_vlan_get_parent(rpcs, interface->if_name, if_par);
+    te_strlcpy(if_name, interface->if_name, IFNAMSIZ);
 
-    if (strlen(if_par) > 0)
-        return TRUE;
+    while (TRUE)
+    {
+        rc = cfg_get_instance_string_fmt(&val,
+                                         "/agent:%s/interface:%s/kind:",
+                                         rpcs->ta, if_name);
+        if (rc != 0)
+        {
+            ERROR("Failed to get kind of %s interface: %r", if_name, rc);
+            return FALSE;
+        }
+        if (strcmp(val, "vlan") == 0)
+        {
+            free(val);
+            return TRUE;
+        }
+        else if (strcmp(val, "") == 0)
+        {
+            free(val);
+            return FALSE;
+        }
+        free(val);
```

Comment:
I'd say better is to have a `te_bool result`, then:
```
if (strcmp(val, "vlan") == 0)
{
     result = TRUE;
     break;
}
```
and in the end, outside of the loop:
```
free(val);
return result;
```

It would make the code logic clearer.